ui/apps/platform/src/Containers/MainPage/InviteUsersConfirmationNoEmail.tsx

```diff
@@ -0,0 +1,101 @@
+import React, { ReactElement } from 'react';
+import { Alert, ClipboardCopy, ClipboardCopyVariant, Text } from '@patternfly/react-core';
+import { Link } from 'react-router-dom';
+
+import { accessControlBasePath } from 'routePaths';
+// eslint-disable-next-line import/no-cycle
+import { EmailBuckets } from './InviteUsersModal';
```

Comment:
Aha, somehow I missed that these files are in MainPage folder.
1. Let’s please move to an InviteUsers subfolder following precedents: Banners, Header, PublicConfig, Sidebar
2. And then is it is natural to factor out cycle imports as something like InviteUsers.ts file.

Reply:
Balancing the cost to look elsewhere for the implementation, with pure helper functions:
* meaningful names (even if naming is hard) make the cause and effect clearer.
* arguments make the inputs obvious and prevent subtle bugs from unintended references to closure if someone else needs to modify the logic later.